glib/src/functions.rs

```diff
@@ -287,3 +287,18 @@ pub fn spawn_future_local<R: 'static, F: std::future::Future<Output = R> + 'stat
     let ctx = crate::MainContext::ref_thread_default();
     ctx.spawn_local(f)
 }
+
+#[doc(alias = "g_compute_checksum_for_string")]
+pub fn compute_checksum_for_string(
+    checksum_type: ChecksumType,
+    str: impl IntoGStr,
+) -> Option<crate::GString> {
+    let length = str.len() as _;
+    unsafe {
+        from_glib_full(ffi::g_compute_checksum_for_string(
+            checksum_type.into_glib(),
+            str.to_glib_none().0,
+            length,
+        ))
+    }
```

Comment:
```suggestion
    unsafe {
        str.run_with_gstr(|str| {
        from_glib_full(ffi::g_compute_checksum_for_string(
            checksum_type.into_glib(),
            str.as_ptr(),
            str.len() as _,
        ))
        })
    }
```

or so. I don't think the current version compiles

Reply:
Thanks, fixed